fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -793,7 +793,8 @@ def _pre_backward_hook(*unused: Any) -> None:
             self._prep_grads_for_backward()
 
         def _register_hook(t: torch.Tensor) -> torch.Tensor:
-            t.register_hook(_pre_backward_hook)
+            if t.requires_grad:
+                t.register_hook(_pre_backward_hook)
```

Comment:
cc @myleott who I think added this

Reply:
No, the `if` is actually a bugfix for modules that may return outputs that don't require grad. In this case pytorch will raise an exception when you call register_hook. I only discovered it because auto_wrap sometimes puts FSDP on module boundaries I hadn't tested before :)